sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/DatabricksShimServiceProvider.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "330db"}
+{"spark": "332db"}
+{"spark": "341db"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids
+
+import scala.util.Try
+
+object DatabricksShimServiceProvider {
+  val log = org.slf4j.LoggerFactory.getLogger(getClass().getName().stripSuffix("$"))
+  def matchesVersion(dbrVersion: String): Boolean = {
+    Try {
+      val sparkBuildInfo = org.apache.spark.BuildInfo
+      val databricksBuildInfo = com.databricks.BuildInfo
+      val matchRes = sparkBuildInfo.dbrVersion.startsWith(dbrVersion)
+      val matchStatus = if (matchRes) "SUCCESS" else "FAILURE"
+      log.warn(s"""Databricks Runtime Build Info match: {}
+                  |\tDBR_VERSION: {}
+                  |\tspark.gitHash: {}
+                  |\tdatabricks.gitHash: {}""".stripMargin,
+        matchStatus,
+        sparkBuildInfo.dbrVersion,
+        sparkBuildInfo.gitHash,
+        databricksBuildInfo.gitHash)
+      matchRes
+    }.recover {
+      case x: Throwable =>
+        log.warn("Databricks detection failed: " + x, x)
```

Comment:
Should this be a debug log?  IIUC we're going to iterate through the shims, calling matchesVersion on each, until we find one that matches.  We don't want a warning appearing on startup for a non-Databricks Spark version, and ShimLoader itself will only log debug when having issues loading any particular shim.